mcu/src/mcu/uart_reader.py

```diff
@@ -0,0 +1,674 @@
+        # Start reading data
+        self.serial_thread.start()
+
+    def disconnect_serial(self):
+        # Check if the serial thread is running
+        if not self.serial_thread:
+            self.logger.error("No serial thread running.")
+            return
+        # Disconnect the serial port
+        try:
+            self.connect_button.clicked.disconnect(self.disconnect_serial)
+            self.connect_button.clicked.disconnect(self.connect_serial)
+        except TypeError:
+            pass
+        self.connect_button.setDisabled(True)
+        # Stop the serial thread
+        self.serial_thread.stop()
+        self.serial_thread.wait()
+        # Update the UI
+        self.status_label.setText("Status: Disconnected")
+        self.status_label.setStyleSheet("color: red;")
+        self.connect_button.setText("Connect")
+        self.connect_button.clicked.connect(self.connect_serial)
+        self.connect_button.setEnabled(True)
+
+    # Update the console output, process the sound buffer and write the audio
+    def update_console(self, message):
+        if message == "TERMINATE":
+            self.disconnect_serial()
+            self.refresh_ports()
+        elif message.startswith(PRINT_PREFIX):
+            # Process the sound buffer
+            data = process_sound_buffer(message)
+            self.aquisition_counter += 1
+            plot_name = f"Audio Plot {self.aquisition_counter}"
+
+            # Log general information
+            self.logger.info(f">> Aquisition {self.aquisition_counter} completed")
+            self.logger.info(
+                f">> Aquisition size of {len(data)} samples encoded in {len(message) - len(PRINT_PREFIX)} bytes"
+            )
+
+            # Plot and write audio
+            plot_data_and_fft(
+                data,
+                self.settings["sampling_frequency"],
+                self.settings["vdd"],
+                self.settings["max_adc_value"],
+                self.settings["plot_output_type"],
+                plot_name,
+                self.logger,
+            )
+            write_audio(
+                data,
+                self.settings["sampling_frequency"],
+                self.settings["audio_output_folder"],
+                f"audio_{self.aquisition_counter}",
+                self.settings["audio_output_type"],
+                self.settings["overwrite_audio"],
+                self.logger,
+            )
+        else:
+            # self.console_output.append(message)  #HACK Covered by the custom handler
+            self.logger.info(f"Received data: {message}")
+
+
+# CLI application logic
+def run_cli(settings):
+    # Set up logging
+    logger = logging.getLogger(__name__)
+
+    # Console information
+    logger.info("\033[92mUART Console App for LELEC210x\033[m")
+    logger.info("\033[32mDeveloped by Group E 2024 - 2025\033[m")
+    logger.info("\033[36mTo exit the application, press Ctrl+C\033[m")
```

Comment:
Don't use special color characters like this, because there are not supported by all terminals, e.g., the `cmd.exe` terminal doesn't recognize them.

If you really want to use color, you can use custom Logging handlers, like [Rich's handler](https://rich.readthedocs.io/en/stable/logging.html), but it seems to me a bit overkill to use that just few three logs.